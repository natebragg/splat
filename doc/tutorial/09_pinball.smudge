// 09_pinball.smudge
pinball
{
   * power-on -(flippers.lock)-> idle,
     idle
     [
          coin --> prepare-ball
     ],
     prepare-ball (@releaseBall, @resetScore, flippers.unlock)
     [
          plunger --> run
     ],
     run (@startSound)
     [
          drain -(@sadSound, @displayScore)-> idle,
          target -(@incScore)-
     ] (flippers.lock),
     lockout (@resetScore, @displayError, @startTimer)
     [
          timerExpired --> idle,
          tilt --,
          _ --
     ] (@cancelTimer),
     _
     [
          tilt --> lockout,
          coin -(@rejectCoin)-
     ]
}

flippers
{
   * locked (@clickLock)
     [
          unlock --> unlocked,
          _ -- // Ignore flipper and lock events.
     ],
     unlocked (@soundFree)
     [
          left -(@flipLeft)-,
          right -(@flipRight)-,
          lock --> locked
     ]
}
