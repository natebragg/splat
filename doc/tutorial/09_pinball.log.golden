Dropping ball.
Ding ding ding ding ding
Coin at a bad time. Dropping it.
Locking flippers.
Flippers away!
fLipDingfLipDingfLipDingfLipDingflipR
Awwwwwww
Score: 4
A new high score!
Dropping ball.
Ding ding ding ding ding
DingDingTILT!
Coin at a bad time. Dropping it.
Locking flippers.
Flippers away!
Locking flippers.
